exchanges/binance/binance_wrapper.go

```diff
@@ -1494,14 +1499,14 @@ func (b *Binance) GetHistoricCandlesExtended(pair currency.Pair, a asset.Item, s
 		Asset:    a,
 		Interval: interval,
 	}
-
-	dates := kline.CalcDateRanges(start, end, interval, b.Features.Enabled.Kline.ResultLimit)
-	for x := range dates {
+	// binance is not inclusive in its ranges, so the end date is not part of the response, the way this breaks up dates ensures that it is not included
+	dates := kline.CalculateCandleDateRanges(start.Add(-interval.Duration()*3), end.Add(interval.Duration()*3), interval, b.Features.Enabled.Kline.ResultLimit)
```

Comment:
I want to start by saying I'm not crazy.... 👀  I did previously encounter an issue where results were not being returned for the last candle. I'll do more tests to see why, otherwise I'll remove the modifications

Reply:
From experimentation it looks like its somewhere in the backtester do I declare missing data when there isn't any. I've reverted this change and will investigate the issue